src/core/Akka.Cluster/Member.cs

```diff
@@ -52,9 +52,9 @@ public static Member Removed(UniqueAddress node)
         readonly ImmutableHashSet<string> _roles;
         public ImmutableHashSet<string> Roles { get { return _roles; } }
 
-        public static Member Create(UniqueAddress uniqueAddress, MemberStatus status, ImmutableHashSet<string> roles)
+        public static Member Create(UniqueAddress uniqueAddress, int upNumber, MemberStatus status, ImmutableHashSet<string> roles)
         {
-            return new Member(uniqueAddress, 0, status, roles);
+            return new Member(uniqueAddress, upNumber, status, roles);
```

Comment:
Oh wow, **nice** catch. I could see this being responsible for some fun issues with Cluster rejoin scenarios too.